roofit/roofitcore/inc/RooAbsMCStudyModule.h

```diff
@@ -34,145 +34,163 @@ class RooAbsMCStudyModule : public TNamed {
-    return kTRUE ; 
+  /// Method called at the end of each RooMCStudy run. If a RooDataSet is returned, it must have a length equal to
+  /// the number of toy experiments performed and will merged with the fitpar dataset of RooMCStudy.
+  virtual RooDataSet* finalizeRun() {
+    return 0 ;
   }
 
+  /// Method called after resetting of generator parameters to initial values and before call to generator context
+  /// Any modifications to generator parameters will affect next generation operation (only)
+  virtual Bool_t processBeforeGen(Int_t /*sampleNum*/) {
+    return kTRUE ;
+  }
 
-  virtual Bool_t processBetweenGenAndFit(Int_t /*sampleNum*/) { 
-    // Method called after generation of toy data sample and resetting of fit parameters to initial values and before
-    // actual fit is performed. Any modifications to fit parameters will apply to next fit operation. Note that setConstant
-    // flag of fit parameters are not explicitly reset by RooMCStudy, so any changes made to these flags here will persist
-    return kTRUE ; 
+  /// Method called after generation of toy data sample and resetting of fit parameters to initial values and before
+  /// actual fit is performed. Any modifications to fit parameters will apply to next fit operation. Note that setConstant
+  /// flag of fit parameters are not explicitly reset by RooMCStudy, so any changes made to these flags here will persist
+  virtual Bool_t processBetweenGenAndFit(Int_t /*sampleNum*/) {
+    return kTRUE ;
   }
 
-  virtual Bool_t processAfterFit(Int_t /*sampleNum*/) { 
-    // Method called after fit has been performed.
-    return kTRUE ; 
+  /// Method called after fit has been performed.
+  virtual Bool_t processAfterFit(Int_t /*sampleNum*/) {
+    return kTRUE ;
   }
 
 protected:
 
-   // Interface methods to RooMCStudy objects, 
+   // Interface methods to RooMCStudy objects,
    // which are only functional after module has been attached to a RooMCStudy object
 
-   RooFitResult* refit(RooAbsData* inGenSample=0) { 
-     // Refit model using orignal or specified data sample
-     if (_mcs) return _mcs->refit(inGenSample) ; else return 0 ; 
+   /// Refit model using orignal or specified data sample
+   RooFitResult* refit(RooAbsData* inGenSample=0) {
+     if (_mcs) return _mcs->refit(inGenSample) ; else return 0 ;
    }
 
-   RooAbsData* genSample() { 
-     // Return generate sample
-     return _mcs ? _mcs->_genSample : 0 ; 
+   /// Return generate sample
+   RooAbsData* genSample() {
+     return _mcs ? _mcs->_genSample : 0 ;
    }
+
+   /// Return generator pdf
    RooAbsPdf* genModel() {
-     // Return generator pdf
-     return _mcs ? _mcs->_genModel : 0 ; 
+     return _mcs ? _mcs->_genModel : 0 ;
+   }
+
+   // Accessor for generator context, generator parameters, prototype data and projected dependents
+
+   /// Return generator context
+   RooAbsGenContext* genContext() {
```

Comment:
```suggestion
   // Accessor for generator context, generator parameters, prototype data and projected dependents.
   RooAbsGenContext* genContext() {
```
There is duplicate documentation now if you move `Return generator context` out like this.